commands/pull_request.go

```diff
@@ -222,6 +232,18 @@ func pullRequest(cmd *Command, args *Args) {
 		utils.Check(fmt.Errorf("Aborting due to empty pull request title"))
 	}
 
+	if flagPullRequestPush {
+		if remote == nil {
+			utils.Check(fmt.Errorf("Can't find remote for %s", head))
+		}
+
+		if args.Noop {
+			args.Before(fmt.Sprintf("Would push to %s/%s", remote.Name, head), "")
+		} else {
+			git.Push(remote.Name, "HEAD", head)
```

Comment:
I've been using this utility personally and in practice I haven't had this race condition bite me, so it would be a shame to delay all requests by multiple seconds when only a few would benefit. I think it's probably worth implementing the retry logic, but I have a couple questions:

* What would the 422 response's body look like?
* How should I structure the retry logic? It looks like only the `Client.CreatePullRequest()` method has access to the status code, but it seems wrong to always retry in that method, since any other call won't be susceptible to this race condition.